ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/Envelope.scala

```diff
@@ -132,17 +132,19 @@ object Envelope {
       case msg => Left(s"Expected state value, got ${msg.getClass}")
     }
 
-  private def compress(payload: ByteString): ByteString = {
+  private[kvutils] def compress(payload: ByteString): ByteString = {
     val out = ByteString.newOutput
     val gzipOut = new GZIPOutputStream(out)
     gzipOut.write(payload.toByteArray)
     gzipOut.close()
     out.toByteString
   }
 
-  private def decompress(payload: ByteString): ByteString = {
+  private[kvutils] def decompress(payload: ByteString): ByteString = {
     val gzipIn = new GZIPInputStream(payload.newInput)
-    ByteString.readFrom(gzipIn)
+    val decompressed = ByteString.readFrom(gzipIn)
+    gzipIn.close()
```

Comment:
We should probably close even if this fails.

```suggestion
    val decompressed = try {
      ByteString.readFrom(gzipIn)
    } finally {
      gzipIn.close()
    }
```

And the same above.

Reply:
Good point -- addressed in #7384.